Makefile

```diff
@@ -27,99 +28,93 @@ endif
+	@${DOCKER_PHP} php -d 'xdebug.mode=off' bin/tools/phpstan/vendor/phpstan/phpstan/phpstan analyse
 
-bin/tools/phpstan/vendor: vendor bin/tools/phpstan/composer.json $(wildcard bin/tools/phpstan/composer.lock)
-	@${DOCKER_PHP} composer bin phpstan install
+bin/tools/phpstan/vendor: bin/tools/phpstan/composer.json bin/tools/phpstan/composer.lock
+	@$(MAKE) --no-print-directory docker-start
+	@$(MAKE) --no-print-directory vendor
+	@$(MAKE) --no-print-directory composer bin phpstan install
 
-database-generate-migration: docker-start vendor database-drop-schema ### Generate new migration based on mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
+database-generate-migration: vendor database-drop-schema ### Generate new migration based on mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
 	@${DOCKER_PHP} vendor/bin/doctrine-migrations migrations:migrate --no-interaction --allow-no-migration # first, let's load into db existing migrations
 	@${DOCKER_PHP} vendor/bin/doctrine-migrations migrations:diff --no-interaction
 	@${DOCKER_PHP} vendor/bin/doctrine-migrations migrations:migrate --no-interaction # load the new migration
 	@${DOCKER_PHP} bin/doctrine orm:validate-schema
 
-database-validate-mapping: docker-start vendor database-drop-schema ### Validate mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
+database-validate-mapping: vendor database-drop-schema ### Validate mapping in Zenstruck\Foundry\Tests\Fixtures\Entity
 	@${DOCKER_PHP} vendor/bin/doctrine-migrations migrations:migrate --no-interaction --allow-no-migration
 	@${DOCKER_PHP} bin/doctrine orm:validate-schema
 
-database-drop-schema: docker-start vendor ### Drop database schema
+database-drop-schema: vendor ### Drop database schema
 	@${DOCKER_PHP} bin/doctrine orm:schema-tool:drop --force
 	@${DOCKER_PHP} vendor/bin/doctrine-migrations migrations:sync-metadata-storage # prevents the next command to fail if migrations table does not exist
 	@${DOCKER_PHP} bin/doctrine dbal:run-sql "TRUNCATE doctrine_migration_versions" --quiet
 
-vendor:  composer.json $(wildcard composer.lock) $(wildcard .env)
-	@${DC_EXEC} -e SYMFONY_REQUIRE=${SYMFONY_REQUIRE} php composer update --prefer-dist
+vendor: composer.json $(wildcard composer.lock) $(wildcard .env)
+	@$(MAKE) --no-print-directory docker-start
+	@${DC_EXEC} -e SYMFONY_REQUIRE=${SYMFONY_REQUIRE} ${PHP} php -d 'xdebug.mode=off' /usr/bin/composer update --prefer-dist
 
-UID = $(shell id -u)
-DOCKER_IS_UP = $(shell $(DOCKER_COMPOSE) ps | grep "Up (healthy)")
+DOCKER_IS_UP = $(shell $(DOCKER_COMPOSE) ps | grep "${PHP}" | grep "Up")
 
 docker-start: ### Build and run containers
-ifeq ($(DOCKER_IS_UP),)
-ifeq ($(CI),'false')
-	@$(DOCKER_COMPOSE) build  --build-arg UID="${UID}" --build-arg XDEBUG_HOST="${XDEBUG_HOST}"
-	@$(DOCKER_COMPOSE) up --detach --remove-orphans
-else
-	@$(DOCKER_COMPOSE) up --detach --no-build
-endif
-	@$(DOCKER_COMPOSE) ps
-endif
+	@if [ -z "${DOCKER_IS_UP}" ]; then \
+		./docker/build.sh load "${PHP_VERSION}" ; \
```

Comment:
we target only the current PHP_VERSION, thus the php images are built/up on demand